### PR TITLE
chats list performance

### PR DESCRIPTION
**Pull request by flexsurfer**
chats list performance

currently, it took 7s to show 1000 chats
1)json is huge
2)json->js is slow (~200ms)
3)js->cljs is slow (~2s)
4)subscriptions are slow, because we reduce all chats few times
5)flatlist is slow 

it's still slow for 1000 chats, but much better
what have been done:
1)decreased json size, by removing group chats data and lastMessage data on go side
2)js->cljs replaced by manually creating of data structures
3)subscriptions improved
4)rendering improved

in this PR it takes ~1-2s to show 1000 chats, which is fine, because it's really rare case, further improvements can be done with JSI

### CODE REVIEW

**Comment by Serhy** [comment]
@flexsurfer there is a regression with chat appearing after account creation/recover for the first time (which mostly makes all above e2e tests fail)
Any chat user joins/creates for the first time after account creation won't appear on the Chats view. However after relogin all those added chats are visible as well as all subsequently added chats.
Similar with Timeline: after account creation the very first post in Timeline won't be visible, but second, third and so on - visible.

**Reproduction**
1) Create new account 
2) Join to any chat (public, create 1-1)
3) Post text message in timeline (Status tab)

**Comment by flexsurfer** [comment]
thanks @Serhy fixed

File: src/status_im/data_store/chats.cljs [hunk 1]
```diff
@@ -63,15 +72,46 @@
...
+       :unviewed-messages-count (.-unviewedMessagesCount chat)
+       :unviewed-mentions-count (.-unviewedMentionsCount chat)
+       :last-message (when (.-lastMessage chat)
+                       {:content {:text (.-lastMessage.text chat)}
+                        :content-type (.-lastMessage.contentType chat)})
+       :last-clock-value (.-lastClockValue chat)
+       :profile-public-key (.-profile chat)
+       :members (.-members chat)}
+      rpc->type
+      (unmarshal-members members-reducer-js)
```
Comment (cammellos) [inline src/status_im/data_store/chats.cljs]: Part of this could be moved to status-go btw if that helps (not in this pr of course)

**Review by cammellos — APPROVED** [review]

**Comment by Serhy** [comment]
@flexsurfer builds are failing on unit tests 
```
[2021-09-13T17:00:25.017Z] FAIL in (normalize-chat-test) (status_im/data_store/chats_test.cljs:42:11)

[2021-09-13T17:00:25.017Z] from-rpc

[2021-09-13T17:00:25.017Z] expected: (= expected-chat (chats/<-rpc chat))

[2021-09-13T17:00:25.017Z]   actual: (not (= {:chat-name "name", :color "color", :contacts #{"d" "a" "b" "c"}, :last-clock-value 10, :admins #{"a" "b"}, :members-joined #{"a" "c"}, :name "name", :last-message nil, :unviewed-messages-count 2, :chat-type 3, :group-chat true, :public? false, :chat-id "chat-id", :timestamp 2, :membership-update-events :events} {:color "color", :last-clock-value 10, :name "name", :last-message nil, :unviewed-messages-count 2, :chat-type 3, :chat-id "chat-id", :timestamp 2, :membership-update-events :events}))
```

**Comment by flexsurfer** [comment]
@Serhy sorry, fixed

**Comment by Serhy** [comment]
Tested with Android 9 and iOS 14. Improvement indeed visible (below videos slowed down 4 times) - with 500 chats on chat view it loads 3 times faster on my device:

PR build:

https://user-images.githubusercontent.com/8749671/133282317-2293d683-5772-4b83-a2b3-50121433af3b.mov

Develop:

https://user-images.githubusercontent.com/8749671/133282344-2288f39b-0e5e-4c28-85a4-58722ff706dc.mov

**Comment by Serhy** [comment]
**Issue 1** 
@flexsurfer nothing appear in Activity Center when received 1-1 or Group chat requests. Indicator of "Something new appeared in Activty center" is present (blue dot on bell icon) - but empty view when you open it. Re-login doesn't help in that case. 

Reproduced on real devices (both iOS and Android)
And from autotests video with logs:

Message sender - > https://app.saucelabs.com/tests/f6fba5577b584d848f0140580dddcf9d?auth=fde17162b4a4e005c7e872f531f73ec3#84

Message receiver: https://app.saucelabs.com/tests/e7d7e76f01524b26928d6b113593371b?auth=00c42121f427555ee1619c7431cc6d1f#65 (also receives Push Notification fine, - but when opens Activity Center - empty view there)

**Comment by flexsurfer** [comment]
thanks @Serhy fixed

**Comment by Serhy** [comment]
Tested on iOS and Android and no regressions found, - above bugs are fixed. 
All failed tests are known issues in Develop (https://github.com/status-im/status-react/issues/12596, https://github.com/status-im/status-react/issues/12586)
Thanks a lot, @flexsurfer !

**Comment by Serhy** [comment]
Still the same good results, ready for merge! Thanks @flexsurfer

**Pull request merged**